index.js

```diff
@@ -1,29 +1,134 @@
 'use strict';
 
-const {Adapter, Device} = require('gateway-addon');
+const {Adapter, Device, Property} = require('gateway-addon');
 const wol = require('wol');
+const findDevices = require('local-devices');
+const {promise: ping} = require('ping');
 
 class WakeOnLanAdapter extends Adapter {
+  static getDeviceInfoFromArpTable(mac, arpDevices = []) {
+    const normalizedMac = mac.toLowerCase();
+    const arpDevice = arpDevices.find((d) => d.mac.toLowerCase() === normalizedMac);
+    if (arpDevice) {
+      // de-normalize mac so device ID stays the same.
+      arpDevice.mac = mac;
+      return arpDevice;
+    }
+    return {
+      mac: mac,
+      name: '?',
+    };
+  }
+
   constructor(addonManager, manifest) {
     super(addonManager, manifest.name, manifest.name);
     addonManager.addAdapter(this);
 
-    for (const mac of manifest.moziot.config.devices) {
-      this.handleDeviceAdded(new WakeOnLanDevice(this, mac));
+    this.checkPing = manifest.moziot.config.checkPing || true;
```

Comment:
This will always end up being `true`. The right thing would be:

```javascript
this.checkPing = manifest.moziot.config.hasOwnProperty('checkPing') ?
  manifest.moziot.config.checkPing :
  true;
```

Reply:
Ugh. Yes, of course.